--- libextractor-0.5.15/src/plugins/Makefile.am.orig	2006-09-16 00:40:25.022938000 +0200
+++ libextractor-0.5.15/src/plugins/Makefile.am	2006-09-16 00:52:22.479776250 +0200
@@ -16,7 +16,7 @@
 endif
 
 if HAVE_QT_SVG
- svgflags = -lQtSvg4
+ svgflags = -lQtSvg
 endif
 
 if HAVE_PRINTABLE
@@ -306,7 +306,7 @@
 libextractor_thumbnailqt_la_LIBADD = \
   $(top_builddir)/src/main/libextractor.la
 libextractor_thumbnailqt_la_LDFLAGS = \
-  $(PLUGINFLAGS) $(retaincommand) -lQtCore4 -lQtGui4 $(svgflags)
+  $(PLUGINFLAGS) $(retaincommand) -lQtCore -lQtGui $(svgflags)
 
 install-exec-hook:
 	install -d $(DESTDIR)$(plugindir) 2> /dev/null
--- libextractor-0.5.15/configure.ac.orig	2006-09-16 00:40:27.623100000 +0200
+++ libextractor-0.5.15/configure.ac	2006-09-16 00:53:04.402396250 +0200
@@ -273,14 +273,14 @@
         ;;
       yes)
         AC_CHECK_HEADERS(qpixmap.h,
-          AC_CHECK_LIB([QtGui4], [_ZN7QPixmap12defaultDepthEv],
+          AC_CHECK_LIB([QtGui], [_ZN7QPixmap12defaultDepthEv],
             qt=1))
         ;;
       *)
         LDFLAGS="-L$with_qt/lib $LDFLAGS"
         CPPFLAGS="-I$with_qt/include -I$with_qt/include/Qt $CPPFLAGS"
         AC_CHECK_HEADERS(qpixmap.h,
-          AC_CHECK_LIB([QtGui4], [_ZN7QPixmap12defaultDepthEv],
+          AC_CHECK_LIB([QtGui], [_ZN7QPixmap12defaultDepthEv],
             EXT_LIB_PATH="-L$with_qt/lib $EXT_LIB_PATH"
             qt=1))
         ;;
@@ -288,7 +288,7 @@
    ],
    [AC_MSG_RESULT([--with-qt not specified])
     AC_CHECK_HEADERS(qpixmap.h,
-     AC_CHECK_LIB([QtGui4], [_ZN7QPixmap12defaultDepthEv],
+     AC_CHECK_LIB([QtGui], [_ZN7QPixmap12defaultDepthEv],
       qt=1))])
 
 if test x$qt != x0
@@ -307,14 +307,14 @@
         ;;
       yes)
         AC_CHECK_HEADERS(qsvgrenderer.h,
-          AC_CHECK_LIB([QtSvg4], [_ZN10QSvgWidgetD0Ev],
+          AC_CHECK_LIB([QtSvg], [_ZN10QSvgWidgetD0Ev],
             qt_svg=1))
         ;;
       *)
         LDFLAGS="-L$with_qt/lib $LDFLAGS"
         CPPFLAGS="-I$with_qt/include -I$with_qt/include/Qt $CPPFLAGS"
         AC_CHECK_HEADERS(qsvgrenderer.h,
-          AC_CHECK_LIB([QtSvg4], [_ZN10QSvgWidgetD0Ev],
+          AC_CHECK_LIB([QtSvg], [_ZN10QSvgWidgetD0Ev],
             EXT_LIB_PATH="-L$with_qt/lib $EXT_LIB_PATH"
             qt_svg=1))
         ;;
@@ -322,7 +322,7 @@
    ],
    [AC_MSG_RESULT([--with-qt not specified])
     AC_CHECK_HEADERS(qsvgrenderer.h,
-     AC_CHECK_LIB([QtSvg4], [_ZN10QSvgWidgetD0Ev],
+     AC_CHECK_LIB([QtSvg], [_ZN10QSvgWidgetD0Ev],
       qt_svg=1))])
 
 if test x$qt_svg != x0
