--- libextractor-0.5.19/src/plugins/Makefile.am.orig	2007-12-31 14:17:46.000000000 +0100
+++ libextractor-0.5.19/src/plugins/Makefile.am	2008-01-22 21:45:54.695968011 +0100
@@ -373,7 +373,7 @@
   $(QT_CFLAGS) $(QT_SVG_CFLAGS)
 
 install-exec-hook:
-	mkdir -p $(DESTDIR)$(plugindir) &> /dev/null || true
+	mkdir -p $(DESTDIR)$(plugindir) 2> /dev/null || true
 	rm -f $(DESTDIR)$(plugindir)/libextractor_thumbnail$(LIBEXT)
 	rm -f $(DESTDIR)$(plugindir)/libextractor_thumbnail.la
 	if test "$(thumbgtk)" != ""; then \
