--- libextractor-1.0.1/configure.ac.orig	2012-12-01 16:51:17.064452494 +0100
+++ libextractor-1.0.1/configure.ac	2012-12-01 16:54:48.781114748 +0100
@@ -458,8 +458,8 @@
 # smf requires glib.h
 CFLAGS_OLD=$CFLAGS
 CPPFLAGS_OLD=$CPPFLAGS
-export CFLAGS="$CFLAGS $GLIB_CFLAGS"
-export CPPFLAGS="$CPPFLAGS $GLIB_CFLAGS"
+CFLAGS="$CFLAGS $GLIB_CFLAGS"
+CPPFLAGS="$CPPFLAGS $GLIB_CFLAGS"
 
 AC_CHECK_LIB(smf, smf_load_from_memory,
         [AC_CHECK_HEADERS([smf.h],
@@ -467,8 +467,8 @@
            AC_DEFINE(HAVE_MPEG2,1,[Have libsmf]),
          AM_CONDITIONAL(HAVE_SMF, false))],
          AM_CONDITIONAL(HAVE_SMF, false))
-export CFLAGS=$CFLAGS_OLD
-export CPPFLAGS=$CPPFLAGS_OLD
+CFLAGS="$CFLAGS_OLD"
+CPPFLAGS="$CPPFLAGS_OLD"
 
 # check for gtk >= 2.6.0
 AC_MSG_CHECKING(for gtk)
