--- libextractor-1.3/src/plugins/gif_extractor.c~	2013-06-21 18:00:16.000000000 +0200
+++ libextractor-1.3/src/plugins/gif_extractor.c	2014-11-10 08:06:50.807376620 +0100
@@ -78,7 +78,7 @@
   if (gif_file == NULL || gif_error != 0)
   {
     if (gif_file != NULL)
-      EGifCloseFile (gif_file);
+      EGifCloseFile (gif_file, NULL);
     return; /* not a GIF */
   }
 #endif
@@ -133,7 +133,7 @@
 	       DGifGetExtensionNext(gif_file, &ext)) &&
 	      (NULL != ext) ) ; /* keep going */
     }
-  DGifCloseFile (gif_file);
+  DGifCloseFile (gif_file, NULL);
 }
 
 /* end of gif_extractor.c */
