diff --git a/ChangeLog b/ChangeLog
index a1ba981..6cd0e63 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,6 @@
+Tue 20 Apr 2021 06:33:46 PM CEST
+    Removing plugins depending on buggy, unstable libffmpeg library. -CG
+
 Sat 30 Jan 2021 03:25:35 PM CET
 	Fix #6709: mask SIGPIPE to avoid test case failure
 	with latest librpm.  Remove plibc.h logic.
diff --git a/configure.ac b/configure.ac
index 456287a..80ce414 100644
--- a/configure.ac
+++ b/configure.ac
@@ -648,38 +648,6 @@ CPPFLAGS="$CPPFLAGS_OLD"
 AC_CHECK_PROG([HAVE_ZZUF],[zzuf], 1, 0)
 AM_CONDITIONAL(HAVE_ZZUF, test 0 != $HAVE_ZZUF)
 
-AC_MSG_CHECKING([whether to enable the FFmpeg thumbnail extractor])
-new_ffmpeg=0
-AC_ARG_ENABLE(ffmpeg,
- [AC_HELP_STRING([--enable-ffmpeg],[Enable FFmpeg support])
-  AC_HELP_STRING([--disable-ffmpeg],[Disable FFmpeg support])],
- [AS_CASE(["$enableval"],
-   [no],[
-      AC_MSG_RESULT(no)
-      ffmpeg_enabled=0
-   ],[
-      AC_MSG_RESULT(yes)
-      ffmpeg_enabled=1
-   ])],
-  [     AC_MSG_RESULT(yes)
-        ffmpeg_enabled=1])
-AS_IF([test x$ffmpeg_enabled = x1],
- [ffmpeg_enabled=0
-  new_ffmpeg=0
-  AC_CHECK_HEADERS([libavutil/avutil.h libavformat/avformat.h libavcodec/avcodec.h libavutil/frame.h],
-    AC_CHECK_HEADERS([libavresample/avresample.h],
-      AC_CHECK_LIB(avutil, av_audio_fifo_alloc,
-        new_ffmpeg=1)))
-
-  AC_CHECK_LIB(swscale, sws_getContext,
-    AC_CHECK_LIB(avcodec, avcodec_alloc_context3,
-      ffmpeg_enabled=1))
-  AC_CHECK_HEADERS([libavutil/avutil.h ffmpeg/avutil.h libavformat/avformat.h ffmpeg/avformat.h libavcodec/avcodec.h ffmpeg/avcodec.h libswscale/swscale.h ffmpeg/swscale.h])
- ])
-AM_CONDITIONAL(HAVE_FFMPEG, test x$ffmpeg_enabled != x0)
-AM_CONDITIONAL(HAVE_FFMPEG_NEW, test x$new_ffmpeg != x0)
-
-
 LE_INTLINCL=""
 LE_LIBINTL="$LTLIBINTL"
 AC_ARG_ENABLE(framework, [  --enable-framework      enable Mac OS X framework build helpers],enable_framework_build=$enableval)
@@ -775,12 +743,6 @@ AS_IF([test "x$HAVE_GIF_TRUE" = "x#"],
 AS_IF([test "x$have_gsf" != "xtrue"],
   [AC_MSG_NOTICE([NOTICE: libgsf not found, no OLE2 (MS Office) support])])
 
-AS_IF([test "x$ffmpeg_enabled" = "x0"],
-  [AC_MSG_NOTICE([NOTICE: FFmpeg thumbnailer plugin disabled])])
-
-AS_IF([test "x$new_ffmpeg" = "x0"],
-  [AC_MSG_NOTICE([NOTICE: FFmpeg/opus audio preview plugin disabled, It needs libav >= 10, or a FFmpeg with --enable-libavresample])])
-
 AS_IF([test "x$have_gdk_pixbuf" = "x0"],
   [AC_MSG_NOTICE([NOTICE: libgdk-pixbuf not found, gtk thumbnail support disabled])])
 
diff --git a/src/plugins/Makefile.am b/src/plugins/Makefile.am
index 8cdd905..69cb1f7 100644
--- a/src/plugins/Makefile.am
+++ b/src/plugins/Makefile.am
@@ -68,17 +71,6 @@ if HAVE_MAGIC
 PLUGIN_MIME=libextractor_mime.la
 TEST_MIME=test_mime
 
-if HAVE_FFMPEG
-# FFmpeg-thumbnailer requires MAGIC and FFMPEG
-PLUGIN_FFMPEG=libextractor_thumbnailffmpeg.la
-TEST_FFMPEG=test_thumbnailffmpeg
-endif
-
-if HAVE_FFMPEG_NEW
-PLUGIN_PREVIEWOPUS=libextractor_previewopus.la
-TEST_PREVIEWOPUS=test_previewopus
-endif
-
 if HAVE_GDK_PIXBUF
 # Gtk-thumbnailer requires MAGIC and GDK_PIXBUF
 PLUGIN_GTK=libextractor_thumbnailgtk.la
@@ -188,7 +182,6 @@ plugin_LTLIBRARIES = \
   libextractor_xm.la \
   $(PLUGIN_ARCHIVE) \
   $(PLUGIN_EXIV2) \
-  $(PLUGIN_FFMPEG) \
   $(PLUGIN_FLAC) \
   $(PLUGIN_GIF) \
   $(PLUGIN_GSF) \
@@ -202,7 +195,6 @@ plugin_LTLIBRARIES = \
   $(PLUGIN_MPEG) \
   $(PLUGIN_OGG) \
   $(PLUGIN_PDF) \
-  $(PLUGIN_PREVIEWOPUS) \
   $(PLUGIN_RPM) \
   $(PLUGIN_TIFF) \
   $(PLUGIN_ZLIB)
@@ -228,8 +222,6 @@ check_PROGRAMS = \
   test_zip \
   $(TEST_ARCHIVE) \
   $(TEST_EXIV2) \
-  $(TEST_FFMPEG) \
-  $(TEST_PREVIEWOPUS) \
   $(TEST_FLAC) \
   $(TEST_GIF) \
   $(TEST_GSF) \
@@ -615,19 +634,6 @@ test_sid_LDADD = \
   $(top_builddir)/src/plugins/libtest.la
 
 
-libextractor_thumbnailffmpeg_la_SOURCES = \
-  thumbnailffmpeg_extractor.c
-libextractor_thumbnailffmpeg_la_LDFLAGS = \
-  $(PLUGINFLAGS)
-libextractor_thumbnailffmpeg_la_LIBADD = \
-  -lavutil -lavformat -lavcodec -lswscale -lmagic $(XLIB)
-
-test_thumbnailffmpeg_SOURCES = \
-  test_thumbnailffmpeg.c
-test_thumbnailffmpeg_LDADD = \
-  $(top_builddir)/src/plugins/libtest.la
-
-
 libextractor_thumbnailgtk_la_SOURCES = \
   thumbnailgtk_extractor.c
 libextractor_thumbnailgtk_la_LDFLAGS = \
@@ -642,18 +648,6 @@ test_thumbnailgtk_SOURCES = \
 test_thumbnailgtk_LDADD = \
   $(top_builddir)/src/plugins/libtest.la
 
-libextractor_previewopus_la_SOURCES = \
-  previewopus_extractor.c
-libextractor_previewopus_la_LDFLAGS = \
-  $(PLUGINFLAGS)
-libextractor_previewopus_la_LIBADD = \
-  -lavutil -lavformat -lavcodec -lswscale -lavresample  -lmagic $(XLIB)
-
-test_previewopus_SOURCES = \
-  test_previewopus.c
-test_previewopus_LDADD = \
-  $(top_builddir)/src/plugins/libtest.la
-
 
 libextractor_tiff_la_SOURCES = \
   tiff_extractor.c
